File: Anul2/Semestrul_1/Metode avansate de programare/Laboratoare/Lab2/lab2_schelet/sem1_2/model/AbstractContainer.java
package sem1_2.model;

import java.util.ArrayList;
import java.util.List;

public abstract class AbstractContainer implements Container {
    List<Task> tasks = new ArrayList<Task>();

    public abstract Task remove();

    public void add(Task task) {
        tasks.add(task);
    }

    public int size() {
        return tasks.size();
    }

    public boolean isEmpty() {
        return tasks.isEmpty();
    }
}

File: Anul2/Semestrul_1/Metode avansate de programare/Laboratoare/Lab9/ConsoleApp1/model/AbstractContainer.cs
namespace ConsoleApp1.model;

public abstract class AbstractContainer : IContainer
{
    protected List<Task> tasks = new List<Task>();

    public abstract Task Remove();

    public void Add(Task task)
    {
        tasks.Add(task);
    }

    public int Size()
    {
        return tasks.Count;
    }

    public bool IsEmpty()
    {
        return Size() == 0;
    }
}